diff --git a/app/B2c/Repositories/Models/User.php b/app/B2c/Repositories/Models/User.php
index 70828f047..88d4ef722 100644
--- a/app/B2c/Repositories/Models/User.php
+++ b/app/B2c/Repositories/Models/User.php
@@ -95,6 +95,7 @@ class User extends BaseModel implements AuthenticatableContract, AuthorizableCon
         'is_ni_nc',
         'created_by',
         'updated_by',
+        'is_review_submit_initiate',
     ];
 
     /**
diff --git a/app/Http/Controllers/Application/ApplicationController.php b/app/Http/Controllers/Application/ApplicationController.php
index 61e503f10..8965c691d 100644
--- a/app/Http/Controllers/Application/ApplicationController.php
+++ b/app/Http/Controllers/Application/ApplicationController.php
@@ -908,6 +908,8 @@ class ApplicationController extends Controller
         Session::put('isReviewSubmit', '1');
         Session::forget('selected_match');
         try {
+            $userinfo['is_review_submit_initiate'] = 1;
+            $this->userRepo->updateUser($userId, $userinfo);
             $userDetails = $this->userRepo->getUserDetail($userId);
             $appData = $this->application->getApplicationDetail($userId, $appId);
             $applicantInfo = $this->application->getApplicantDetailsByID($appId, config('b2c_common.MAIN_APPLICANT'));
diff --git a/app/Http/Controllers/Auth/RegisterController.php b/app/Http/Controllers/Auth/RegisterController.php
index 9ae4bc013..0f62607a2 100644
--- a/app/Http/Controllers/Auth/RegisterController.php
+++ b/app/Http/Controllers/Auth/RegisterController.php
@@ -372,13 +372,25 @@ class RegisterController extends Controller
                 $activationUserInfo = $this->activationRepo->getUserInfoByToken($token);
                 if(!empty($activationUserInfo) && $activationUserInfo->is_verified == 1){
                     if(!empty($login_user_id)){
-                        return redirect(route('in_progress_application'))->with('message', 'Your email link has been already verified.');
+                        /*if (!empty($app_id) && ( Session::has('isReviewSubmit') && Session::get('isReviewSubmit')==1 )) {*/
+                        if (!empty($app_id) && ($userDataLogin->is_review_submit_initiate==1 )) {
+                            $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                            return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]))->with('message', 'Your email link has been already verified.');
+                        } else {
+                            return redirect(route('in_progress_application'))->with('message', 'Your email link has been already verified.');
+                        }
                     }else{
                         return redirect($this->loginPath())->with('message', 'Your email link has been already verified.');
                     }
                 }else{
                     if(!empty($login_user_id)){
-                        return redirect(route('in_progress_application'))->withErrors(['Your email link has been expired.']);
+                        /*if (!empty($app_id) && ( Session::has('isReviewSubmit') && Session::get('isReviewSubmit')==1 )) {*/
+                        if (!empty($app_id) && ($userDataLogin->is_review_submit_initiate==1 )) {
+                            $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                            return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]))->withErrors(['Your email link has been expired.']);
+                        } else {
+                            return redirect(route('in_progress_application'))->withErrors(['Your email link has been expired.']);
+                        }
                     }else{
                         return redirect($this->loginPath())->withErrors(['Your email link has been expired.']);
                     }
@@ -399,20 +411,35 @@ class RegisterController extends Controller
      * @return  redirect
      */
     public function verifyUserEditedEmail($token)
-    {
+    {   
         $userData = Auth::user();
+
         $user_id = (isset($userData) && !empty($userData)) ? $userData->id : '';
         $tokenValid = 0;
         $msName = 'messageVerify';
-     
+        $app_id = null;
+        $applicant_id = null;
+        $lastApp = [];
+
         try {
             $userData = $this->activationRepo->getUserInfoByToken($token);
+            $userDetails = $this->userRepo->getUserDetail($user_id);
             if (!empty($user_id)) {
+                $lastApp = $this->application->getLatestApplicationByStatus($user_id, config('b2c_common.INCOMPLETE_APP'));
+                $app_id = (isset($lastApp) && !empty($lastApp)) ? $lastApp->app_id :'';
+                $applicant_id = (isset($lastApp) && !empty($lastApp)) ? $lastApp->app_applicant_id : '';
                 $tokenValid = Helpers::checkTokenValidForUser($user_id, $token);
                 $msName = 'message';
                 if ($tokenValid <= 0) {
                     if(!empty($userData) && $userData->is_verified == 1){
-                        return redirect(route('edit_profile'))->with('message', 'Your email link has been already verified.');
+                        if($userDetails->is_review_submit_initiate == 1){
+                            $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                            return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]))->with('message', 'Your email link has been already verified.');
+                        }
+                        else{
+                            return redirect(route('edit_profile'))->with('message', 'Your email link has been already verified.');
+                        }
+                        
                     } else {
                         return redirect(route('edit_profile'))->withErrors(['Your email link has been expired.']);
                     }
@@ -443,20 +470,41 @@ class RegisterController extends Controller
                     }
                 Session::flash($msName, 'Your email has been verified.');
                 if (!empty($user_id)) {
-                    return redirect(route('edit_profile'));
+                    if($userDetails->is_review_submit_initiate == 1){
+                        $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                        return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]));
+                    }
+                    else{
+                        return redirect(route('edit_profile'));
+                    }
+                    
                 } else {
                     return redirect($this->loginPath()); 
                 }
             } else {
                 if(!empty($userData) && $userData->is_verified == 1){
                     if (!empty($user_id)) {
-                        return redirect(route('edit_profile'))->with('message', 'Your email link has been already verified.');
+                        if($userDetails->is_review_submit_initiate == 1){
+                            $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                            return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]))->with('message', 'Your email link has been already verified.');
+                        }
+                        else{
+                            return redirect(route('edit_profile'))->with('message', 'Your email link has been already verified.');
+                        }
+                        
                     } else {
                         return redirect($this->loginPath())->with('message', 'Your email link has been already verified.');
                     } 
                 } else {
                    if (!empty($user_id)) {
-                        return redirect(route('edit_profile'))->withErrors(['Your email link has been expired.']);
+                        if($userDetails->is_review_submit_initiate == 1){
+                            $this->userRepo->updateUser($user_id,['is_review_submit_initiate'=>0]);
+                            return redirect(route('review_submit', ['user_id' => $user_id, 'app_id' => $app_id, 'applicant_id' => $applicant_id]))->withErrors(['Your email link has been expired.']);
+                        }
+                        else{
+                            return redirect(route('edit_profile'))->withErrors(['Your email link has been expired.']);
+                        }
+                        
                     } else {
                         return redirect($this->loginPath())->withErrors(['Your email link has been expired.']);
                     }  
