diff --git a/app/B2c/Repositories/Entities/Application/ApplicationRepository.php b/app/B2c/Repositories/Entities/Application/ApplicationRepository.php
index 868a1fcd4..d82e50b93 100644
--- a/app/B2c/Repositories/Entities/Application/ApplicationRepository.php
+++ b/app/B2c/Repositories/Entities/Application/ApplicationRepository.php
@@ -1204,4 +1204,14 @@ class ApplicationRepository extends BaseRepositories implements ApplicationInter
     {
         return LenderStatus::getOrderByStatusId($StatusId);
     }
+
+    /**
+     * Get user latest application details
+     * 
+     * @param type $user_id
+     * @return array
+     */
+    public static function getUserLatestAppDetails($user_id){
+        return Application::getUserLatestAppDetails($user_id);
+    }
 }
\ No newline at end of file
diff --git a/app/Http/Controllers/Auth/AccountController.php b/app/Http/Controllers/Auth/AccountController.php
index 8077f5ca0..13bd94128 100644
--- a/app/Http/Controllers/Auth/AccountController.php
+++ b/app/Http/Controllers/Auth/AccountController.php
@@ -348,15 +348,44 @@ class AccountController extends Controller
      * @param Request $request
      */
     public function verifyEmail(Request $request)
-    {
+    {   
          try {
+            $user_id        = Auth::user()->id;
             $userData = $this->userRepo->getLastEmailByUserID(\Auth::user()->id);
+            $userAppDetails = $this->application->getUserLatestAppDetails(Auth::user()->id);
+
+            if($userAppDetails){
+                $arrPersonalInfo = $this->application->getApplicantDetailsByID($userAppDetails['app_id'], config('b2c_common.MAIN_APPLICANT'));
+                $applicant_id = ($arrPersonalInfo['app_applicant_id'])?$arrPersonalInfo['app_applicant_id']:null;
+                //dd(config('b2c_common.CO_APPLICANT'));
+                $coApplicant = $this->application->getApplicantDetailsByID($userAppDetails['app_id'], config('b2c_common.CO_APPLICANT'));
+                if($coApplicant == false){
+                    $coApplicant = null;
+                }
+            }else{
+                $coApplicant = null;
+                $applicant_id = null;
+            }
+
+            if($userAppDetails){
+                $userAppDetails = $this->application->getApplicationDetail($user_id, $userAppDetails['app_id']);
+                $currentStage = $userAppDetails->app_stage_no;
+            }else{
+                $currentStage = 0;
+            }
+
             if(!$userData && empty($userData)){
                 Session::flash('message', trans('messages.email_edited_success'));
                 return Redirect::to(route('edit_profile'));
             }
             return view('frontend.account.edit_email_verify')
-                    ->with('userData', $userData);
+                    ->with('userData', $userData)
+                    ->with('user_id', $user_id)
+                    ->with('userAppDetails',$userAppDetails)
+                    ->with('coApplicant', $coApplicant)
+                    ->with('currentStage', $currentStage)
+                    ->with('applicant_id', $applicant_id);
+;
             
         } catch (\Exception $ex) {
             return redirect()->back()->withErrors(Helpers::getExceptionMessage($ex));
diff --git a/app/Http/Controllers/Backend/DashboardController.php b/app/Http/Controllers/Backend/DashboardController.php
index 74a8a632e..510c30614 100644
--- a/app/Http/Controllers/Backend/DashboardController.php
+++ b/app/Http/Controllers/Backend/DashboardController.php
@@ -705,13 +705,38 @@ class DashboardController extends Controller
      * @param Request $request
      */
     public function verifyCustomerEmail(Request $request)
-    {
+    {   
          try {
             $user_id        = (int)request()->get('user_id');
+            $userAppDetails = $this->application->getUserLatestAppDetails($user_id);
+            
+            if($userAppDetails){
+                $arrPersonalInfo = $this->application->getApplicantDetailsByID($userAppDetails['app_id'], config('b2c_common.MAIN_APPLICANT'));
+                $applicant_id = ($arrPersonalInfo['app_applicant_id'])?$arrPersonalInfo['app_applicant_id']:null;
+                $coApplicant = $this->application->getApplicantDetailsByID($userAppDetails['app_id'], config('b2c_common.CO_APPLICANT'));
+                if($coApplicant == false){
+                    $coApplicant = null;
+                }
+            }else{
+                $coApplicant = null;
+                $applicant_id = null;
+            }
+
+            if($userAppDetails){
+                $userAppDetails = $this->application->getApplicationDetail($user_id, $userAppDetails['app_id']);
+                $currentStage = $userAppDetails->app_stage_no;
+            }else{
+                $currentStage = 0;
+            }
+
             $userData       = $this->userRepo->getLastEmailByUserID($user_id);
             return view('backend.account.edit_customer_email_verify')
                     ->with('userData', $userData)
-                    ->with('user_id', $user_id);
+                    ->with('user_id', $user_id)
+                    ->with('userAppDetails',$userAppDetails)
+                    ->with('coApplicant', $coApplicant)
+                    ->with('currentStage', $currentStage)
+                    ->with('applicant_id', $applicant_id);
 
         } catch (\Exception $ex) {
             return redirect()->back()->withErrors(Helpers::getExceptionMessage($ex));
diff --git a/resources/views/backend/account/edit_customer_email_verify.blade.php b/resources/views/backend/account/edit_customer_email_verify.blade.php
index eb6bcd9e2..e5dba2939 100644
--- a/resources/views/backend/account/edit_customer_email_verify.blade.php
+++ b/resources/views/backend/account/edit_customer_email_verify.blade.php
@@ -30,9 +30,39 @@
                                 <a href="{{ route('edit_customer_profile',['user_id' => $user_id]) }}" target="_blank" class="hvr-wobble-vertical btnemail">Modify customer Email-id</a>
                         </div>
                     </div>
-                    <div class="row goDash">
-                        <div class="col-xs-12 col-md-12 col-lg-12 col-xl-12 text-center"> <a href="{{ route('edit_customer_profile',['user_id' => $user_id]) }}">&lt; Back</a> </div>
+
+            @if(empty($userAppDetails))
+                <div class="row">
+                    <div class="col-xs-12 text-center">
+                        <a href="{{route('backend_pl_loan_detail',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id']])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
                     </div>
+                </div>
+            @else
+                @if($userAppDetails['is_review_submit_completed'] == 1)
+                        <div class="row goDash">
+                                <div class="col-xs-12 col-md-12 col-lg-12 col-xl-12 text-center"> <a href="{{ route('edit_customer_profile',['user_id' => $user_id]) }}">&lt; Back</a></div>
+                        </div>
+                @else
+                    <div class="row">
+                        <div class="col-xs-12 text-center">
+                        @if($currentStage == 1)
+                            <a href="{{route('backend_personal_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id ])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 2)
+                            <a href="{{route('backend_applicant_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 3)
+                            <a href="{{route('backend_employment_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 4)
+                            <a href="{{route('backend_financial_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 5 && !empty($coApplicant))
+                            <a href="{{route('backend_co_applicant_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @else
+                            <a href="{{route('backend_review_submit',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @endif
+                        </div>
+                    </div>  
+                @endif
+            @endif
+                    
 
 
             </div>
diff --git a/resources/views/frontend/account/edit_email_verify.blade.php b/resources/views/frontend/account/edit_email_verify.blade.php
index 1dd1cccf2..d9cbb773f 100644
--- a/resources/views/frontend/account/edit_email_verify.blade.php
+++ b/resources/views/frontend/account/edit_email_verify.blade.php
@@ -34,9 +34,38 @@
                             <a href="{{route('edit_profile')}}" target="_blank" class="hvr-wobble-vertical btnemail">Modify your Email-id</a>
                         </div>
                     </div>
-                    <div class="row goDash">
-                        <div class="col-xs-12 col-md-12 col-lg-12 col-xl-12 text-center"> <a href="{{route('edit_profile')}}">&lt; Back</a> </div>
+                    
+            @if(empty($userAppDetails))
+                <div class="row">
+                    <div class="col-xs-12 text-center">
+                        <a href="{{route('pl_loan_detail',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id']])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                    </div>
+                </div>
+            @else
+                @if($userAppDetails['is_review_submit_completed'] == 1)
+                        <div class="row goDash">
+                            <div class="col-xs-12 col-md-12 col-lg-12 col-xl-12 text-center"> <a href="{{ route('edit_profile',['user_id' => $user_id]) }}">&lt; Back</a></div>
+                        </div>
+                @else
+                    <div class="row">
+                        <div class="col-xs-12 text-center">
+                        @if($currentStage == 1)
+                            <a href="{{route('personal_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id ])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 2)
+                            <a href="{{route('applicant_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 3)
+                            <a href="{{route('employment_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 4)
+                            <a href="{{route('financial_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @elseif($currentStage == 5 && !empty($coApplicant))
+                            <a href="{{route('co_applicant_info',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @else
+                            <a href="{{route('review_submit',['user_id' => $user_id, 'app_id' => $userAppDetails['app_id'], 'applicant_id' => $applicant_id])}}" class="btn btn-sm btnAction btn-primary"></em> Continue </a>
+                        @endif
+                        </div>
                     </div>
+                    @endif
+                @endif
                     
                
             </div>
